src/Phoenix/PhoenixClient.ts

```diff
@@ -25,25 +25,40 @@ export class PhoenixClient {
   public async containerConnectionInfo(
     provisionData: IProvosionData
   ): Promise<IPhoenixResponse<IPhoenixConnectionInfoResult>> {
-    const response = await window.fetch(`${this.getPhoenixContainerPoolingEndPoint()}/provision`, {
-      method: "POST",
-      headers: PhoenixClient.getHeaders(),
-      body: JSON.stringify(provisionData),
-    });
-    let data: IPhoenixConnectionInfoResult;
-    if (response.status === HttpStatusCodes.OK) {
-      data = await response.json();
-    } else {
-      const connectionStatus: DataModels.ContainerConnectionInfo = {
+    try {
+      const connectionStatus: ContainerConnectionInfo = {
+        status: ConnectionStatusType.Connecting,
+      };
+      useNotebook.getState().setConnectionInfo(connectionStatus);
```

Comment:
> useNotebook.getState() [](http://example.com/codeflow?start=5&length=23)

do we really want state as part of phoenix client?

why dont phoenix client truly be just a client?

Reply:
will address this in V2